projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js

```diff
@@ -40,6 +41,30 @@
 				}
 			} );
 		}
+
+		if ( adminMenu ) {
+			var collapseButton = adminMenu.querySelector( '#collapse-button' );
+			// Nav-Unification feature:
+			// Saves the sidebar state in server when "Collapse menu" is clicked.
+			// This is needed so that we update WPCOM for this preference in real-time.
+			if ( collapseButton ) {
+				collapseButton.addEventListener( 'click', function ( event ) {
+					// Let's the core event listener be triggered first.
+					setTimeout( function () {
+						saveSidebarIsExpanded( event.target.parentNode.ariaExpanded );
+					}, 50 );
+				} );
+			}
+		}
+	}
+
+	function saveSidebarIsExpanded( expanded ) {
+		var xhr = new XMLHttpRequest();
+		xhr.open( 'POST', '/wp-admin/admin-ajax.php', true );
```

Comment:
Non-blocking: I think you can use the WP utils for Ajax requests here: https://codex.wordpress.org/AJAX_in_Plugins#Separate_JavaScript_File

Reply:
This code uses jQuery and we are making efforts of removing it p4TIVU-8wx-p2. I tried to follow suit. I replaced though `/wp-admin/admin-ajax.php` with global `ajaxurl` which I wouldn't have found without your comment! Thanks! d2b081c

Reply:
oh, good point, you can ignore then my comment in https://github.com/Automattic/jetpack/pull/19520#discussion_r613872065.